streams/src/test/java/org/apache/kafka/test/MockApiProcessor.java

```diff
@@ -46,14 +47,22 @@ public class MockApiProcessor<KIn, VIn, KOut, VOut> implements Processor<KIn, VI
     private Cancellable scheduleCancellable;
 
     private final PunctuationType punctuationType;
+    private final Instant startTime;
     private final long scheduleInterval;
 
     private boolean commitRequested = false;
     private ProcessorContext<KOut, VOut> context;
 
     public MockApiProcessor(final PunctuationType punctuationType,
                             final long scheduleInterval) {
+        this(punctuationType, null, scheduleInterval);
+    }
+
+    public MockApiProcessor(final PunctuationType punctuationType,
+                            final Instant startTime,
```

Comment:
Same consideration here to make the `startTime` an `Optional` or use `Instant.EPOCH_TIME` as a sentinel value when calling as line 58 above